### PR TITLE
Fix: Disown opener for external links

### LINKED ISSUE #174: External links should have `rel="noopener noreferrer"`
Ref: https://sonarwhal.com/docs/user-guide/rules/disown-opener.html

### PR DESCRIPTION
**Pull request by qzhou1607**
Fix #174

### CODE REVIEW

File: src/hexo/themes/documentation/scripts/external-link-filter.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+            $(this).attr('rel', 'noopener noreferrer'); //eslint-disable-line no-invalid-this
+        }
+    });
+
+    data.content = $.html();
+};
+
+hexo.extend.filter.register('after_post_render', updateRelAttribute, 11);
+// We need to modify the string genereated from a hexo default filter : https://github.com/hexojs/hexo/blob/master/lib/plugins/filter/after_post_render/external_link.js.
+// So we have to execute `updateRelAttribute` after it runs.
```
Comment (molant) [inline src/hexo/themes/documentation/scripts/external-link-filter.js]: This comment should go above

File: src/hexo/themes/documentation/scripts/external-link-filter.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/* globals hexo */
+const cheerio = require('cheerio');
+const updateRelAttribute = (data) => {
+    const $ = cheerio.load(data.content, { decodeEntities: false });
+
+    $('a').each(function () {
+        if ($(this).attr('href') && $(this).attr('rel') === 'external') { //eslint-disable-line no-invalid-this
```
Comment (molant) [inline src/hexo/themes/documentation/scripts/external-link-filter.js]: Disable the eslint rule at the file level. No point in having it in two lines in a row for the amount of code we have here.

Comment (alrra) [inline src/hexo/themes/documentation/scripts/external-link-filter.js]: Fixed.

File: src/hexo/themes/documentation/scripts/external-link-filter.js (hunk 1 above)
Comment (alrra) [inline src/hexo/themes/documentation/scripts/external-link-filter.js]: Fixed.

File: src/hexo/themes/documentation/scripts/external-link-filter.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+            $(this).attr('rel', 'noopener noreferrer'); //eslint-disable-line no-invalid-this
```
Comment (alrra) [inline src/hexo/themes/documentation/scripts/external-link-filter.js]: @qzhou1607 We shouldn't overwrite the existing value.

This should be something such as:

```js
$(this).attr('rel', (i, currentValue) => {
    return [...new Set(['noopener', 'noreferrer'].concat(currentValue.split(' ')))].join(' ');
});      
```

Comment (qzhou1607) [inline src/hexo/themes/documentation/scripts/external-link-filter.js]: @alrra What is the reason to convert the array to a set first before joining?

Comment (alrra) [inline src/hexo/themes/documentation/scripts/external-link-filter.js]: Not having duplicates.

**Pull request closed without merging**